File: lab6.cpp
#include <iostream>
#include <string>

using namespace std;

int call_count = 0;
double sum = 0.0;


void func(int n) {
    call_count += 1;
    
    sum += 1.0 / n;
    
    // Base case
    if (n == 1) {
        std::cout << "Summation : " << sum << std::endl;
        return;
    }
    
    func(n - 1);
}

void func(){
    call_count = 0;
    sum = 0.0;

    int n;
    std::cout << "Enter a positive integer: ";
    std::cin >> n;
    
    if (n <= 0) {
        std::cout << "Please enter a positive integer!" << std::endl;
        return;
    }
    
    func(n);
    
    std::cout << "Number of calls: " << call_count << std::endl;

}

int main() {
    func(5);
    
    std::cout << "Number of calls: " << call_count << std::endl;
    
    return 0;
}

int main() {return 0;}

File: lab6.py
call_count = 0
sum = 0.0

def func(n):
    global call_count, sum
    call_count += 1

    sum += 1.0 / n

    """Base case"""
    if(n == 1):
        print(f"Summation : {sum}")
        return
    
    func(n-1)


def factorial(x):
    if x < 0:
        raise ValueError("Factorial is not defined for negative numbers")
    if x == 0 or x == 1:
        return 1
    
    return x * factorial(x - 1)